Compilation fails when using GCC 13: `error: ignoring '#pragma GCC novector'`

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Affected Build(s)

Latest git compile as of 12:00 27/06/2025 UTC
Latest release (2122.1, azahar-unified-source-2122.1.tar.xz)

### Description of Issue

I am getting this error when I try to compile on arm64:

```
/home/random/git/azahar/src/common/math_util.cpp:68: error: ignoring ‘#pragma GCC novector’ [-Werror=unknown-pragmas]
   68 |     DISABLE_VECTORIZE
      |
/home/random/git/azahar/src/common/math_util.cpp:130: error: ignoring ‘#pragma GCC novector’ [-Werror=unknown-pragmas]
  130 |     DISABLE_VECTORIZE
      |
```

### Expected Behavior

It should compile correctly

### Reproduction Steps

Follow the build guide, at the build step it fails at around 55-65%

### Log File

Note: this is the second time I ran the build because the scrollback was too small.

[log.txt](https://github.com/user-attachments/files/20946258/log.txt)

### System Configuration

![Image](https://github.com/user-attachments/assets/d59cf260-1280-4540-84a6-cb5a3d41a6fc)

--- a/src/common/math_util.cpp
+++ b/src/common/math_util.cpp
@@ -19,7 +19,7 @@
 #define DISABLE_VECTORIZE __pragma(loop(no_vector))
 #elif defined(__clang__)
 #define DISABLE_VECTORIZE _Pragma("clang loop vectorize(disable)")
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) && (__GNUC__ >= 14)
 #define DISABLE_VECTORIZE _Pragma("GCC novector")
 #else
 #define DISABLE_VECTORIZE

